### ISSUE TITLE
Subscription attempts fail with inappropriate message when a connection to the mail service cannot be established

### ISSUE THREAD

**Issue by RadoslavGatev** [TryGhost/Ghost#11401]
### Issue Summary

Subscription attempt fails with an inappropriate message when mail sending service is not configured properly and does not work. Basically what happens is that Ghost attempts to send an email and for example, the authentication to the email service fails. Then it displays: "Please enter a valid email address!"
![image](https://user-images.githubusercontent.com/3099248/69010071-3f346780-0964-11ea-9629-9f27aebd677b.png)

### To Reproduce

1. Set a wrong password for the mail service for example
2. Try subscribing to the blog with a valid email address.

### Technical details:

* Ghost Version: 3.0.3
* Node Version: 10.16.3
* Browser/OS: Windows
* Database: sqlite

**Comment by gargol** [comment]
Hi @RadoslavGatev :wave: Thanks for the report. This error message definitely could be improved and is due to a very generic error handling which is implemented on subscribe form at this moment in [Casper](https://github.com/TryGhost/Casper/blob/b51dd8c/partials/subscribe-form.hbs#L16) 
The same problem can be found in other variations on the members-specific theme [Lyra](https://github.com/TryGhost/Lyra/blob/beace3cf11ae246972a7c9cbed00802fa5a244ff/partials/subscribe-form.hbs#L16). 

Was initially thinking there could be something improved on the theme side, but seems like the main code for handling subscribe request and eventual error handling is in main Ghost repository:
https://github.com/TryGhost/Ghost/blob/master/core/server/public/members.js#L29-L38

**Comment by oreniko26** [comment]
I just experienced the same problem only I had a different sender address configured in the config file than the one configured under labs.
It took me a while to understand what was happening but by the end of the process I realized I can't have one address configured to the newsletter feature (which sometimes I expect people to reply to) and the sign in email which I never want to deal with replies to.
Should I open a new ticket on this?
